controllers/Pokemons.js

```diff
@@ -0,0 +1,124 @@
+const models = require('../models')
+
+const getAllPokemon = async (request, response) => {
+  try {
+    const allPokemon = await models.Pokemons.findAll()
+
+    return response.send(allPokemon)
+  } catch (error) {
+    return response.status(500).send('Could not retrieve all pokemon, please try again.')
+  }
+}
+
+const getPokemonById = async (request, response) => {
+  try {
+    const { id } = request.params
+    const pokemon = await models.Pokemons.findOne({
+      where: { name: id },
+      include: [
+        { model: models.Forms },
+        { model: models.Types }
+      ]
+    })
+
+    return pokemon
+      ? response.status(200).send(pokemon)
+      : response.status(404).send(`Could not find the pokemon ${id}.`)
+  } catch (error) {
+    return response.status(500).send('Could not retrieve the pokemon, please try again.')
+  }
+}
+
+const getPokemonByGenerationId = async (request, response) => {
+  try {
+    const { id } = request.params
+    const allPokemonInGeneration = await models.Pokemons.findAll({ where: { generationNumber: id } })
+
+    return allPokemonInGeneration.length
+      ? response.send(allPokemonInGeneration)
+      : response.status(404).send(`No generation ${id} pokemon currently exists.`)
+  } catch (error) {
+    return response.status(500).send('Could not retrieve the pokemon by the sepcified generation, please try again.')
+  }
+}
+
+const saveNewPokemon = async (request, response) => {
+  try {
+    const {
+      name,
+      generationNumber,
+      fromId,
+      types
+    } = request.body
+    const isProtected = request.body.isPortected || 0
+
+
+    if (!name || !generationNumber) {
+      return response
+        .status(400)
+        .send('At least one of the following attributes is missing: pokedexNumber, name, or generationNumber')
+    }
+
+    const [savedPokemon, created] = await models.Pokemons.findOrCreate({
+      where: { name },
+      defaults: { generationNumber, fromId, isProtected }
```

Comment:
You need to clean up the fromId, you can't insert the string `'null'` because it is a string not null.

![image](https://user-images.githubusercontent.com/5775716/83667262-1db36a00-a59c-11ea-9d2a-8275083937a5.png)